app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,81 @@
+class PostsController < ApplicationController
+  before_action :post_params, only: [:create, :update]
+  before_action :find_post, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!, except: [:show, :index]
+  before_action :authorize!, only: [:edit, :update, :delete]
+
+  PER_PAGE = 10
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.user = current_user
+    if @post.save
+      redirect_to post_path(@post), notice: "Post created!"
+    else
+      flash[:alert] = "See errors below:"
+      render :new
+    end
+  end
+
+  def show
+    respond_to do |format|
+      @comment = Comment.new
+      @favourite = @post.favourites.find_by_user_id(current_user.id) if user_signed_in?
+      format.html { render }
+      format.json { render json: @comment }
+    end
+  end
+
+  def edit
+    #redirect_to root_path, alert: "Access Denied!" unless can? :edit, @post
+  end
+
+  def update
+    current_post = Post.find(params[:id])
+    unless params[:post][:favourite].nil?
```

Comment:
Just wanted to put a marker in here to ask you why you included updating favourites in the update post controller.